src/Symfony/Component/Yaml/Parser.php

```diff
@@ -336,7 +336,8 @@ private function getNextEmbedBlock($indentation = null, $inSequence = false)
                 $removeComments = !preg_match($removeCommentsPattern, $this->currentLine);
             }
 
-            if ($isItUnindentedCollection && !$this->isStringUnIndentedCollectionItem($this->currentLine)) {
+            $isStringUnIndentedShortcutKey = (2 === ($indent - $newIndent));
```

Comment:
@xabbuh event it not related to this PR, here is another issue:

```yaml
collection:
    -  key: foo
```

Excepted:
```json
{
  "collection": [
    {
      "key": "foo"
    }
  ]
}
```

But given:
```json
{
  "collection": [
    {
        "key: foo"
    }
  ]
}
```